### DISCUSSION TITLE
underlines  are replaced with spaces in navigation bar

### DISCUSSION THREAD

**Discussion by xie-dd** [squidfunk/mkdocs-material#7271] [Q&A] [answered]
My markdown file name contains underline, but the underline in the navigation bar of the generated web page are replaced with spaces. Is there something I did not set correctly?
Any help is appreciated.
![image](https://github.com/squidfunk/mkdocs-material/assets/148318644/b94f4e61-98e3-4e65-a7fe-58ef0f532923)

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @xie-dd,
this is the expected behaviour set by MkDocs:
- https://github.com/mkdocs/mkdocs/blob/fb1d106747930b094bc464189d4b1c349fc541c6/mkdocs/structure/pages.py#L227-L261

As per the code you should be able to override it via setting the `title` property in the front matter to the one containing the underscores if they're important `title: Calu_WeiXin...`, otherwise it processes the file name and replaces the `_` with ` `.
- https://squidfunk.github.io/mkdocs-material/reference/#setting-the-page-title

If you want to change the default behaviour you'd need to override the Python code with a hook, but I'm not sure if that's a good approach🤔 Another option is to add a hook that would override the default name later on without patching the internal code.
- https://www.mkdocs.org/user-guide/configuration/#hooks

**Reply by kamilkrzyskow** [reply]
Answered in mkdocs discussions:
- https://github.com/mkdocs/mkdocs/discussions/3758
